src/datamodel/validator/Uuid.cpp

```diff
@@ -10,3 +10,134 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/validator/Uuid.h"
+
+using namespace std::chrono_literals;
+
+namespace {
+std::chrono::system_clock::time_point getUuidTimestamp(
+    const uprotocol::v1::UUID& uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t timestamp =
+	    (msb >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
```

Comment:
I'd recommend using constants (or constexpr's) for a lot of these bitmasks and shift values.  Alternatively create a macro that you can feed the relavant pieces into.

Reply:
Thank you, I'll add these suggestions, and following. It should improve readability